File: in_Python/0657 Robot Return to Origin.py
class Solution(object):
    def judgeCircle(self, moves):
        """
        :type moves: str
        :rtype: bool
        """
        res = 0
        for i in range(len(moves)):
            if moves[i] == 'U':
                res += 1
            elif moves[i] == 'D':
                res -= 1
            elif moves[i] == 'L':
                res += 2
            else:
                res -= 2
        return res == 0
        
        

File: in_Java/0657 Robot Return to Origin.java
class Solution {
    public boolean judgeCircle(String moves) {
        char[] ch = moves.toCharArray();
        int x = 0;
        int y = 0;
        for(char c : ch){
            if(c == 'U') x += 1;
            else if (c == 'D') x -= 1;
            else if (c == 'L') y -= 1;
            else y += 1;
        }
        return x == 0 && y == 0;
    }
}